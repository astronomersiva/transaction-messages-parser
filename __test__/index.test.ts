const { getTransactionData } = await import('../src/index.js');
import fs from 'fs';

const testCases = fs.readFileSync('__test__/inputs').toString().split('-----\n');

test('main', async() => {
  for (const testCase of testCases) {
    expect(await getTransactionData(testCase)).toMatchSnapshot();
  }
});
