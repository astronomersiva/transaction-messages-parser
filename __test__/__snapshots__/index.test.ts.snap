// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
{
  "account": {
    "name": null,
    "number": "x5203",
    "number_formatted": "x5203",
    "type": "card",
  },
  "transaction": {
    "amount": 141.12,
    "currency": "INR",
    "date": "Sun Feb 25 2024 16:41:20 GMT+0530 (India Standard Time)",
    "merchant": "Swiggy",
    "type": "debit",
  },
}
`;

exports[`main 2`] = `
{
  "account": {
    "name": null,
    "number": "x5203",
    "number_formatted": "x5203",
    "type": "card",
  },
  "transaction": {
    "amount": 141,
    "currency": "INR",
    "date": "Sun Feb 25 2024 16:41:20 GMT+0530 (India Standard Time)",
    "merchant": "Swiggy",
    "type": "debit",
  },
}
`;

exports[`main 3`] = `
{
  "account": {
    "name": null,
    "number": "xx4540",
    "number_formatted": "x4540",
    "type": "card",
  },
  "transaction": {
    "amount": 959.1,
    "currency": "INR",
    "date": "Tue Dec 19 2023 15:07:06 GMT+0530 (India Standard Time)",
    "merchant": "PAYPAL *CLOUDFLARE",
    "type": "debit",
  },
}
`;

exports[`main 4`] = `
{
  "account": {
    "name": null,
    "number": "x4540",
    "number_formatted": "x4540",
    "type": "card",
  },
  "transaction": {
    "amount": 2191,
    "currency": "INR",
    "date": "Thu Feb 15 2024 21:58:55 GMT+0530 (India Standard Time)",
    "merchant": "BUNDL TECHNOLOGIES PVT",
    "type": "debit",
  },
}
`;

exports[`main 5`] = `
{
  "account": {
    "name": null,
    "number": "xx4540",
    "number_formatted": "x4540",
    "type": "card",
  },
  "transaction": {
    "amount": 25,
    "currency": "INR",
    "date": "Sat Nov 11 2023 07:13:58 GMT+0530 (India Standard Time)",
    "merchant": "TRAVEL CLUB LOUNGE",
    "type": "debit",
  },
}
`;

exports[`main 6`] = `
{
  "account": {
    "name": null,
    "number": "XXXXXX5913",
    "number_formatted": "x5913",
    "type": "upi",
  },
  "transaction": {
    "amount": 11,
    "currency": "INR",
    "date": "Tue Feb 20 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "goog-payment@okaxis",
    "type": "credit",
  },
}
`;

exports[`main 7`] = `
{
  "account": {
    "name": null,
    "number": "**5913",
    "number_formatted": "x5913",
    "type": "upi",
  },
  "transaction": {
    "amount": 890,
    "currency": "INR",
    "date": "Sun Feb 11 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "Shell India Markets Pvt Ltd",
    "type": "debit",
  },
}
`;

exports[`main 8`] = `
{
  "account": {
    "name": null,
    "number": "*5913",
    "number_formatted": "x5913",
    "type": "upi",
  },
  "transaction": {
    "amount": 23,
    "currency": "INR",
    "date": "Sun Feb 25 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "VARUN MIDDLENAME LASTNAME",
    "type": "debit",
  },
}
`;

exports[`main 9`] = `
{
  "account": {
    "name": null,
    "number": "XX5489",
    "number_formatted": "x5489",
    "type": "card",
  },
  "transaction": {
    "amount": 12345,
    "currency": "INR",
    "date": "Wed Jan 31 2024 20:05:04 GMT+0530 (India Standard Time)",
    "merchant": "DREAMPLUG T",
    "type": "debit",
  },
}
`;

exports[`main 10`] = `
{
  "account": {
    "name": null,
    "number": "x0613",
    "number_formatted": "x0613",
    "type": "card",
  },
  "transaction": {
    "amount": 1500,
    "currency": "INR",
    "date": "Sat Feb 17 2024 06:59:46 GMT+0530 (India Standard Time)",
    "merchant": "NEELKANTH PETROLEUM",
    "type": "debit",
  },
}
`;

exports[`main 11`] = `
{
  "account": {
    "name": null,
    "number": "x0613",
    "number_formatted": "x0613",
    "type": "card",
  },
  "transaction": {
    "amount": 2500,
    "currency": "INR",
    "date": "Fri Feb 16 2024 19:07:26 GMT+0530 (India Standard Time)",
    "merchant": "ABC CORP CAPTIVE",
    "type": "debit",
  },
}
`;

exports[`main 12`] = `
{
  "account": {
    "name": null,
    "number": "XX8167",
    "number_formatted": "x8167",
    "type": "upi",
  },
  "transaction": {
    "amount": 305,
    "currency": "INR",
    "date": "Sat Jan 06 2024 15:06:10 GMT+0530 (India Standard Time)",
    "merchant": "HOTEL NEW ARCOT",
    "type": "debit",
  },
}
`;

exports[`main 13`] = `
{
  "account": {
    "name": null,
    "number": "XX8167",
    "number_formatted": "x8167",
    "type": "upi",
  },
  "transaction": {
    "amount": 305,
    "currency": "INR",
    "date": "Sat Jan 06 2024 15:06:10 GMT+0530 (India Standard Time)",
    "merchant": "HOTEL NEW ARCOT",
    "type": "debit",
  },
}
`;

exports[`main 14`] = `
{
  "account": {
    "name": null,
    "number": "XX4000",
    "number_formatted": "x4000",
    "type": "card",
  },
  "transaction": {
    "amount": 1516,
    "currency": "INR",
    "date": "Wed Feb 21 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "IND*Amazon",
    "type": "debit",
  },
}
`;

exports[`main 15`] = `
{
  "account": {
    "name": null,
    "number": "XX8167",
    "number_formatted": "x8167",
    "type": "upi",
  },
  "transaction": {
    "amount": 23000,
    "currency": "INR",
    "date": "Fri Aug 18 2023 11:55:29 GMT+0530 (India Standard Time)",
    "merchant": "SATHISHKU/HDFC BANK",
    "type": "credit",
  },
}
`;

exports[`main 16`] = `
{
  "account": {
    "name": null,
    "number": "XX187",
    "number_formatted": "x187",
    "type": "upi",
  },
  "transaction": {
    "amount": 123,
    "currency": "INR",
    "date": "Wed Feb 28 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "SATHISHKUMAR PA",
    "type": "debit",
  },
}
`;

exports[`main 17`] = `
{
  "account": {
    "name": null,
    "number": "XX187",
    "number_formatted": "x187",
    "type": "upi",
  },
  "transaction": {
    "amount": 870,
    "currency": "INR",
    "date": "Tue Oct 17 2023 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "Mr BALASURYA P",
    "type": "credit",
  },
}
`;

exports[`main 18`] = `
{
  "account": {
    "name": null,
    "number": "XX187",
    "number_formatted": "x187",
    "type": "upi",
  },
  "transaction": {
    "amount": 2500,
    "currency": "INR",
    "date": "Thu Jul 20 2023 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "S SRIKOWSIKAA",
    "type": "credit",
  },
}
`;

exports[`main 19`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 123456.78,
    "currency": "INR",
    "date": null,
    "merchant": "Salary",
    "type": "credit",
  },
}
`;

exports[`main 20`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 12345,
    "currency": "INR",
    "date": null,
    "merchant": "Salary",
    "type": "credit",
  },
}
`;

exports[`main 21`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 50000,
    "currency": "INR",
    "date": "Sun Dec 03 2023 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "AYYAPPAN A",
    "type": "credit",
  },
}
`;

exports[`main 22`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 12345,
    "currency": "INR",
    "date": "Tue Oct 24 2023 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "ZERODHA BROKING LTD",
    "type": "credit",
  },
}
`;

exports[`main 23`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 3121,
    "currency": "INR",
    "date": "Sun Feb 05 2023 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "xxxxxxxxxxx9426",
    "type": "debit",
  },
}
`;

exports[`main 24`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 3405,
    "currency": "INR",
    "date": null,
    "merchant": null,
    "type": "debit",
  },
}
`;

exports[`main 25`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 1933.5,
    "currency": "INR",
    "date": null,
    "merchant": null,
    "type": "debit",
  },
}
`;

exports[`main 26`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 15000,
    "currency": "INR",
    "date": "Thu Oct 26 2023 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "xxxxxxxxxxx2621",
    "type": "debit",
  },
}
`;

exports[`main 27`] = `
{
  "account": {
    "name": null,
    "number": "XX5913",
    "number_formatted": "x5913",
    "type": "account",
  },
  "transaction": {
    "amount": 5000,
    "currency": "INR",
    "date": "Thu Feb 08 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "ACH D- Groww Pay Services P",
    "type": "debit",
  },
}
`;

exports[`main 28`] = `
{
  "account": {
    "name": null,
    "number": "x4540",
    "number_formatted": "x4540",
    "type": "card",
  },
  "transaction": {
    "amount": 199,
    "currency": "INR",
    "date": "Tue Feb 27 2024 23:13:23 GMT+0530 (India Standard Time)",
    "merchant": "NETFLIX",
    "type": "debit",
  },
}
`;

exports[`main 29`] = `
{
  "account": {
    "name": null,
    "number": "XX8198",
    "number_formatted": "x8198",
    "type": "account",
  },
  "transaction": {
    "amount": 112345,
    "currency": "INR",
    "date": "Wed Feb 28 2024 00:00:00 GMT+0530 (India Standard Time)",
    "merchant": "MYCOMPANY INTERNET PRIVATE LIMITED",
    "type": "credit",
  },
}
`;
